graphql/appointment/fields.ts

```diff
@@ -0,0 +1,188 @@
+import { AuthorizedDeferred, Role, hasAccess } from '../authorizations';
+import { Arg, Authorized, Ctx, Field, FieldResolver, Int, ObjectType, Query, Resolver, Root } from 'type-graphql';
+import { Lecture as Appointment } from '../generated';
+import { GraphQLContext } from '../context';
+import { getSessionStudent, getUserForSession, isElevated, isSessionStudent } from '../authentication';
+import { Deprecated } from '../util';
+import { LimitEstimated } from '../complexity';
+import { prisma } from '../../common/prisma';
+import { getUserIdTypeORM, getUserTypeAndIdForUserId } from '../../common/user';
+
+@ObjectType()
+class AppointmentParticipant {
+    @Field((_type) => String, { nullable: true })
+    userId: string;
+    @Field((_type) => Int, { nullable: true })
+    id: number;
+    @Field((_type) => String, { nullable: true })
+    firstname: string;
+    @Field((_type) => String, { nullable: true })
+    lastname: string;
+    @Field((_type) => Boolean, { nullable: true })
+    isStudent?: boolean;
+    @Field((_type) => Boolean, { nullable: true })
+    isPupil?: boolean;
+    @Field((_type) => Boolean, { nullable: true })
+    isScreener?: boolean;
+}
+
+@ObjectType()
+class Organizer {
+    @Field((_type) => String, { nullable: true })
+    userId: string;
+    @Field((_type) => Int)
+    id: number;
+    @Field((_type) => String)
+    firstname: string;
+    @Field((_type) => String)
+    lastname: string;
+    @Field((_type) => Boolean)
+    isStudent: boolean;
+}
+
+@Resolver((of) => Appointment)
+export class ExtendedFieldsLectureResolver {
+    @Query((returns) => Appointment)
+    @AuthorizedDeferred(Role.OWNER, Role.APPOINTMENT_PARTICIPANT)
+    async appointment(@Ctx() context: GraphQLContext, @Arg('appointmentId') appointmentId: number) {
+        const appointment = await prisma.lecture.findUniqueOrThrow({ where: { id: appointmentId } });
+        await hasAccess(context, 'Lecture', appointment);
+        return appointment;
+    }
+
+    @Deprecated('use isOrganizer instead')
```

Comment:
Should we introduce a new query which is deprecated already?

Reply:
This is an old resolver, but I renamed the file from lectures to appointments.

Reply:
Okay I see.
Interesting that github doesn't show it 😅